rs/execution_environment/src/scheduler/tests.rs

```diff
@@ -4781,50 +4775,64 @@ fn break_after_long_executions(#[strategy(2..10_usize)] scheduler_cores: usize)
 ///    finish the paused execution and should not start any new executions.
 #[test]
 fn filter_after_long_executions() {
+    let scheduler_cores = 2;
     let max_instructions_per_slice = SchedulerConfig::application_subnet()
         .max_instructions_per_slice
         .get();
 
     let mut test = SchedulerTestBuilder::new()
         .with_scheduler_config(SchedulerConfig {
+            scheduler_cores,
             max_instructions_per_round: (max_instructions_per_slice * 2).into(),
             max_instructions_per_message: (max_instructions_per_slice * 2).into(),
             max_instructions_per_message_without_dts: max_instructions_per_slice.into(),
+            max_instructions_per_slice: max_instructions_per_slice.into(),
             ..SchedulerConfig::application_subnet()
         })
         .build();
 
-    // Create a canister with long messages
+    // Create scheduler cores + 1 canisters with long executions.
     let mut long_message_ids = vec![];
     let long_canister_id = test.create_canister();
     for _ in 0..2 {
         let long_message_id =
             test.send_ingress(long_canister_id, ingress(max_instructions_per_slice + 1));
         long_message_ids.push(long_message_id);
     }
+    for _ in 0..scheduler_cores {
+        let another_canister_id = test.create_canister();
+        for _ in 0..2 {
+            test.send_ingress(another_canister_id, ingress(max_instructions_per_slice + 1));
+        }
+    }
+
+    // Remember the initial accumulated priority.
+    let canister = test.state().canister_state(&long_canister_id).unwrap();
+    let ap_before_execution = canister.scheduler_state.accumulated_priority;
 
     // After the first round the canister should have a paused long execution.
     test.execute_round(ExecutionRoundType::OrdinaryRound);
-    for canister in test.state().canisters_iter() {
-        assert_eq!(canister.system_state.canister_metrics.executed, 1);
-        assert!(canister.has_paused_execution());
-    }
+    let canister = test.state().canister_state(&long_canister_id).unwrap();
+    assert_eq!(canister.system_state.canister_metrics.executed, 1);
+    assert!(canister.has_paused_execution());
 
-    // After the second round the canister should have no executions, i.e. the
-    // finish the paused execution and should not start any new executions.
+    // After the second round the canister should finish the paused execution.
     test.execute_round(ExecutionRoundType::OrdinaryRound);
-    for canister in test.state().canisters_iter() {
-        assert_eq!(canister.system_state.canister_metrics.executed, 2);
-        assert!(!canister.has_paused_execution());
-    }
+    let canister = test.state().canister_state(&long_canister_id).unwrap();
+    assert_eq!(canister.system_state.canister_metrics.executed, 2);
+    assert!(!canister.has_paused_execution());
+
+    // The accumulated priority should not increase after the long execution is finished.
```

Comment:
Is this a change in behaviour compared to current state?

Reply:
There are no changes in this test other than the sum of accumulated priorities are zero only within the `apply_accumulated_priority` function, and not zero in the test.

Reply:
I reverted the change here: https://github.com/dfinity/ic/pull/1757/commits/d14b7464a9f981f194143daf29a706bc8afe3f31